experimental/traffic-portal/src/app/core/origins/detail/origin-detail.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HarnessLoader } from "@angular/cdk/testing";
+import { TestbedHarnessEnvironment } from "@angular/cdk/testing/testbed";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { MatButtonHarness } from "@angular/material/button/testing";
+import { MatDialogModule } from "@angular/material/dialog";
+import { MatDialogHarness } from "@angular/material/dialog/testing";
+import { ActivatedRoute } from "@angular/router";
+import { RouterTestingModule } from "@angular/router/testing";
+import { ReplaySubject } from "rxjs";
+
+import { OriginService } from "src/app/api";
+import { APITestingModule } from "src/app/api/testing";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+import { OriginDetailComponent } from "./origin-detail.component";
+
+describe("OriginDetailComponent", () => {
+	let component: OriginDetailComponent;
+	let fixture: ComponentFixture<OriginDetailComponent>;
+	let route: ActivatedRoute;
+	let paramMap: jasmine.Spy;
+	let loader: HarnessLoader;
+	let service: OriginService;
+
+	const navSvc = jasmine.createSpyObj([], {
+		headerHidden: new ReplaySubject<boolean>(),
+		headerTitle: new ReplaySubject<string>(),
+	});
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
+			declarations: [OriginDetailComponent],
+			imports: [APITestingModule, RouterTestingModule, MatDialogModule],
```

Comment:
The reason your tests are failing is because the dialog never actually closes in time for the tests, and things waiting for it to close never happen. The problem is you're waiting on animations, but without explicitly providing an animations module. As you can see in [the testing example in the Angular Material Dialog component documentation](https://material.angular.io/components/dialog/examples#dialog-harness), you should provide the `NoopAnimationsModule` from `@angular/platform-browser/animations`.

Reply:
Thanks @ocket8888. I could make it works